### PR TITLE
fix: deal with reactor without products correctly

### PR DESCRIPTION
**Pull request by lpatiny**

### CODE REVIEW

File: src/com/actelion/research/gwt/core/JSReactor.java
Comment (targos) [inline src/com/actelion/research/gwt/core/JSReactor.java:1]: I think I told you already in a previous PR. If you modify the java source directly, your changes will be lost next time we update the library.

File: src/com/actelion/research/gwt/core/JSReactor.java [hunk 1]
```diff
@@ -23,6 +23,10 @@ public boolean setReactant(int no, JSMolecule reactant) {
   public JSMolecule[][] getProducts() {
     StereoMolecule[][] products = reactor.getProducts();
 
+    if (products == null || products.length == 0) {
+      return null;
+    }
```
Comment (targos) [inline src/com/actelion/research/gwt/core/JSReactor.java:28]: Shouldn't we return an empty array like the original code when we get an empty array?

Comment (lpatiny) [reply src/com/actelion/research/gwt/core/JSReactor.java:28]: I would like but it is an array of array and if we return JSMolecule[] it is another type. I didn't know how to solve this issue.

**Comment by targos** [comment]
You forgot to update the TS typings.

**Pull request merged**